### ISSUE TITLE
BraveBrowserStandaloneSilentSetup.exe unusable with automated install system because of %LOCALAPPDATA%%

### ISSUE THREAD

**Issue by x3nb63** [brave/brave-browser#18711]
## Description 

as it appears the installer uses `%LOCALAPPDATA%\BraveSoftware` as destination.

When a actual user calls the installer that will resolve to `C:\Users\<USERNAME>\AppData\Local` -> OK.

But when a automated install system - which runs as a system service - calls it,  that path resolves to `C:\Windows\System32\config\systemprofile\AppData\Local\` ... then it installs fine, but actual users are unable to find nor access the installation...

## Steps to Reproduce

   1. call the installer from a system service

## Actual result:

installation ends up in the systemprofile, see path above.

## Expected result:

The installer should have a way to specify the destination path, so that a system wide installation in @C:\Program Files\@ becomes possible.

(or at least to a specified user directory, in case the installation is not multi-user capable).

## Reproduces how often: 

always

## Brave version (brave://version info)

1.30.89

## Version/Channel Information:

- Can you reproduce this issue with the current release? 
- Can you reproduce this issue with the beta channel? 
- Can you reproduce this issue with the nightly channel? 

probably yes to all. I have seen it with the mentioned version and with some recent nightly build.

## Other Additional Information:

Its being discussed on the Chocolatey package about 1 year ago. See https://community.chocolatey.org/packages/brave

- Is the issue reproducible on the latest version of Chrome? 

Not aware there is a silent installer.

**Comment by Techy444** [comment]
Just a little update (if it is of any help):

I originally created the "Install Brave Browser For All Windows Users (Ansible + Chocolatey) #25571 submission.

1) If I download the installer for brave online and login as domain admin user and manually install, it will install Brave on all users for that PC.

2) When I login as domain admin and run choco commands manually to install, Brave they will install on all users for that PC. 

3) When I install via Ansible + chocolatey, Brave will only install on the domain admin account and not on any other users for that PC.

Happy to do any tests you want in the future

**Comment by Techy444** [comment]
Update: Logging into domain admin account on windows 10, then running command: choco install brave doesn't install for all users, it will only install for the domain admin account

**Comment by mherrmann** [comment]
I just stumbled across https://github.com/brave/omaha/blob/3fd37b43dad820f01f7cfab761986f9b88cb0eb1/build_omaha.py#L154 and suspect it may be the reason for this unwanted behavior. (We probably want `needsadmin=prefers` instead of `needsadmin=False`).

**Comment by x3nb63** [comment]
thanks. Meanwhile I gave up on using Brave.

**Comment by Neustradamus** [comment]
There is a problem with user profile which is not in the good folder!

Linked to:
- https://github.com/brave/brave-browser/issues/1473
- https://github.com/brave/brave-browser/issues/4609

**Comment by MrMase** [comment]
At the moment also cannot switch to Brave due to this issue, by default silent install should be for all users in program files location, alternatively provide an MSI release that allows for this behaviour which has been requested several times before and makes rollout a lot simplier both persoanlly and in enterprise deployment suites where the silent install commands are simply /qn /norestart (with the latter likely not needed in the majority of cases).

If an MSI is out of the question then at least provide details on supported switches so the default install directory can be defined manually, one of the reasons I like software compiled using the INNO suite as their installers supported switches are well documented.

**Comment by mherrmann** [comment]
Per https://github.com/brave/brave-browser/issues/33084#issuecomment-1723828408, you can invoke the following in an admin command prompt to achieve silent system-wide installation:

    BraveBrowserStandaloneSetup /silent /install

**Comment by omeldar** [comment]
Is there any documentation regarding this topic with the current options/params for BraveBrowserStandaloneSilentSetup.exe? Do I need the params `/silent` and `/install` with that .exe too?

**Comment by mherrmann** [comment]
I'm not sure. Why don't you just use `BraveBrowserStandaloneSetup`?

**Comment by Neustradamus** [comment]
@mherrmann: It is possible to respect Windows Standards?

2022-12-31 (more two years now), I have done this comment:
- https://github.com/brave/brave-browser/issues/18711#issuecomment-1368147010

Linked to:
- https://github.com/brave/brave-browser/issues/1473
- https://github.com/brave/brave-browser/issues/4609

**Comment by mherrmann** [comment]
@Neustradamus if I understand correctly, then the issues and comments you linked to are about where the profile and cache data are stored, and not about the install location.

**Comment by MrMase** [comment]
I would say it’s about both, if Program Files are stored in user profiles this creates bloat and difficulty managing system deployments that is why other browsers such as Firefox, Edge and Chrome honour the use of Program Files.
LocalAppData and Appdata as previously commented have different use cases.
________________________________

**Comment by DDQW** [comment]
This also annoys me quite a lot. Brave does not seperate profile and programm cleanly. Profile of course belongs into the userfolder but none of the programfiles do.

**Comment by Adhjie** [comment]
I don't use Brave CLI installation params directly but mine works using winget.

--machine for system wide
I forgot the user param just search it in Microsoft docs for winget. SS64 website is also fairly good for windows related CLI params.

Admin right usually is needed when installing as system wide.

--interactive is for GUI option.

Silent is just without -i
Or you could suppress it more, I haven't looked much into total silent install.

---

GUI frontend for some package managers, for this case, I use winget:
https://github.com/marticliment/UniGetUI

The manifest for the current latest Brave Browser app version:

https://github.com/microsoft/winget-pkgs/blob/master/manifests/b/Brave/Brave/137.1.79.126/Brave.Brave.installer.yaml

Params from UniGetUI, the GUI app for winget, and many more package managers:

winget.exe install --id "Brave.Brave" --exact --source winget --accept-source-agreements --disable-interactivity --scope machine --interactive  --accept-package-agreements --force --architecture x64

(Disable interactivity is for super silent I think, since no -i already means silent install)
For other params explanation and options, just search it in ms docs.

To search the manifest and make sure the params used in UniGetUI or CLI (Windows Terminal in my case) are correct:

https://github.com/microsoft/winget-pkgs?search=1

Edit:
I haven't tried manual install and the interactive option still just pop up the install window without install wizard, so IG there's no GUI install for Brave. I haven't got the time to test it manually.

Edit 2:
To check for Brave installation folder, and Brave profile folder use brave://version

**Comment by MrMase** [comment]
The cleaner solution here would be to provide an MSI or appropriate variables for use for system administrators to use in deployment scenarios like Microsoft Endpoint Management Console or InTune when deploying a Win32 App for shared devices.

Both of these implemented properly would also resolve the issue in the title of this thread in regards to the use of %LocalAppData% for storing the application in areas other users cannot access as they do not have permission over the install directory.

Winget is a nice idea in theory but doesn't allow the required method of control for example what about a situation where the station needs to be built without an internet connection, installs applications before being made available to the outside world such as HTTPS Inspection Certificates from a task sequence after appropriate station patching.

**Comment by Adhjie** [comment]
@MrMase 
I agree for enterprise settings, Brave browser team should be more attentive.

I think it's possible to cache Chocolatey locally before using it for private enterprise use.
But that will not work indeed, if Brave devs don't give the tool to do this install at an enterprise lvl, silently or interactively, system-wide, or user-wide.

But at least my comment still works for personal install.

**Comment by Neustradamus** [comment]
It is not specified to Enterprise. An user would like to install in C:\Program Files\Brave... easily.

For example, Pulsar does it!
- https://github.com/pulsar-edit/pulsar
- https://github.com/pulsar-edit/pulsar/releases

**Comment by Adhjie** [comment]
@Neustradamus 
(hmm, it only has chocolatey option, thats fine really, at least it has an option, nice:
https://pulsar-edit.dev/download.html#regular-releases
)
Didn't my comment solve this issue for user (thus partially solved)?
Brave does install in \Program Files\ if you install it using winget with these params: --scope machine and launch it with admin right terminal, sudo in 24h2, or use admin option in unigetui.
Brave dev still doesn't give any customization tho, so --interactive only pop up the chrome wizard without any wizard to do it --interactively like FF, etc.
here is my directory:
C:\Program Files\BraveSoftware\Brave-Browser\Application\brave.exe
